firewood/src/merkle/node.rs

```diff
@@ -451,87 +444,83 @@ impl Storable for Node {
     fn serialized_len(&self) -> u64 {
         Meta::SIZE as u64
             + match &self.inner {
-                NodeType::Branch(n) => {
-                    // TODO: add path
-                    n.serialized_len()
-                }
+                NodeType::Branch(n) => n.serialized_len(),
                 NodeType::Extension(n) => n.serialized_len(),
                 NodeType::Leaf(n) => n.serialized_len(),
             }
     }
 
     fn serialize(&self, to: &mut [u8]) -> Result<(), ShaleError> {
         trace!("[{self:p}] Serializing node");
-        let mut cur = Cursor::new(to);
-
-        let mut attrs = match self.root_hash.get() {
-            Some(h) => {
-                cur.write_all(&h.0)?;
-                NodeAttributes::ROOT_HASH_VALID
-            }
-            None => {
-                cur.write_all(&[0; 32])?;
-                NodeAttributes::empty()
-            }
+        let mut cursor = Cursor::new(to);
+
+        let root_hash = self.root_hash.get().map(|hash| hash.0);
+        let encoded = self.encoded.get();
+        let encoded_len = encoded.map(Vec::len).map(|len| len as u64);
+
+        let mut attrs = match (encoded_len, self.is_encoded_longer_than_hash_len.get()) {
+            // the raw encoded-node is longer than a hash
+            (None, Some(true)) => NodeAttributes::LONG,
+            (Some(len), _) if len > TRIE_HASH_LEN as u64 => NodeAttributes::LONG,
+            //
+            (Some(_), _) | (None, Some(false)) => NodeAttributes::IS_ENCODED_BIG_VALID,
```

Comment:
@xinifinity, do you know what this is used for? I know the attribute is used to set `is_encoded_longer_than_hash_len` here, https://github.com/ava-labs/firewood/blob/bdb82d704b8a305023185b42221cc146ed6ebdf9/firewood/src/merkle/node.rs#L412-L419

But it seems like that's only checked in two places:
 - https://github.com/ava-labs/firewood/blob/bdb82d704b8a305023185b42221cc146ed6ebdf9/firewood/src/merkle/node/branch.rs#L153-L170
  - https://github.com/ava-labs/firewood/blob/bdb82d704b8a305023185b42221cc146ed6ebdf9/firewood/src/merkle/node/extension.rs#L55-L68
  
  But it seems we could just use the `Long` attribute in that case instead.

Reply:
To clarify, it's apparent from this `match` statement that either the `Long` flag is set OR the `IS_ENCODED_BIG_VALID`, but never both. Can we just get rid of it?